server/sonar-server/src/main/java/org/sonar/server/telemetry/TelemetryUrl.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.telemetry;
+
+import org.sonar.api.config.Configuration;
+
+import static org.sonar.server.telemetry.TelemetryProperties.PROP_URL;
+
+class TelemetryUrl {
```

Comment:
IMO it's overkill to have classes TelemetryUrl and TelemetryFrequency. There are both used in only a single place. Moving them to private methods of TelemetryClient and TelemetryDaemon is much simpler, especially because the later classes inject only Configuration.